src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,158 @@ def test_set_beam_shift(self):
+        stigmator_range = self.microscope.stigmator_info()['range']
+        min_x_stig_values, max_x_stig_values = stigmator_range['x']
+        min_y_stig_values, max_y_stig_values = stigmator_range['y']
+        self.assertGreaterEqual(min(stigmator_values_x), min_x_stig_values)
+        self.assertLessEqual(max(stigmator_values_x), max_x_stig_values)
+        self.assertGreaterEqual(min(stigmator_values_y), min_y_stig_values)
+        self.assertLessEqual(max(stigmator_values_y), max_y_stig_values)
+
+        # TODO K.K. remove debug values after testing
+        print("X list is")
+        print(stigmator_values_x)
+        print("Y list is")
+        print(stigmator_values_y)
+        print("With a range of:")
+        print(self.microscope.stigmator_info()['range'])
+
+        # Test if an error is raised if an invalid detector is provided
+        with self.assertRaises(KeyError):
+            self.scanner.applyAutoStigmator("error_expected")
+        time.sleep(2.5)  # Give microscope/simulator the time to update the state
+        autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
+        self.assertEqual(autostigmator_state, False)  # Check if state remained unchanged
+
+    def test_apply_auto_contrast_brightness(self):
+        """
+        Test for the auto functionality of the apply_auto_contrast_brightness.
+        Only checks the returned state, the return object and if an error is raised for an incorrect input state.
+        """
+        for detector in detector2ChannelName:
+            auto_contrast_brightness_return = self.scanner.applyAutoContrastBrightness(detector)
+            time.sleep(2.5)  # Give microscope/simulator the time to update the state
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector])
+            self.assertEqual(auto_contrast_brightness_state, True)
+            self.assertIsInstance(auto_contrast_brightness_return, ProgressiveFuture)
+
+            auto_contrast_brightness_return.cancel()
+            time.sleep(5.0)  # Give microscope/simulator the time to update the state
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector])
+            self.assertEqual(auto_contrast_brightness_state, False)
+            self.assertIsInstance(auto_contrast_brightness_return, ProgressiveFuture)
+
+            self.scanner.applyAutoContrastBrightness(detector)
+            starting_time = time.time()
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            while self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector]):
+                time.sleep(0.2)  # Wait until the autofocus is finished
+                if time.time() - starting_time > 30:
+                    break
+            # Check if the time to perform the autofocus is not to long
```

Comment:
to -> too (This is a mistake I see very often. Please look up how to use "to" and "too").